app/controllers/chapters_controller.rb

```diff
@@ -97,7 +97,7 @@ def edit
   end
 
   def draft_flash_message(work)
-    flash[:notice] = work.posted ? ts("This is a draft chapter in a posted work. It will be kept unless the work is deleted.") : ts("This is a draft chapter in an unposted work. The work will be <strong>automatically deleted</strong> on #{view_context.time_in_zone(work.created_at + 1.month)}.").html_safe
+    flash[:notice] = work.posted ? t("chapters.draft_flash.posted_work") : t("chapters.draft_flash.unposted_work_html", deletion_date: view_context.date_in_zone(work.created_at + 29.days)).html_safe
```

Comment:
Since the key [ends with `_html`](https://guides.rubyonrails.org/i18n.html#using-safe-html-translations), you might not need `.html_safe` here.

Reply:
Unfortunately, the code for HTML-safe translation in controllers was [first introduced in Rails 7.0](https://github.com/rails/rails/commit/fee61e3abc878e4d4930a0be0accf8e4569009a8). So we have a ways to go before we can use that.

Reply:
Whoops, good catch, thanks!